chapters/en/slides/module6/module6_02.Rmd

```diff
@@ -0,0 +1,352 @@
+
+```{python}
+(alt.Chart(state_map).mark_geoshape().transform_lookup(
+    lookup='id',
+    from_=alt.LookupData(state_pop, 'id', ['population']))
+.encode(color='population:Q')
+.project(type='albersUsa'))
+```
+
+Notes:
+We could perform the linkage manually,
+by adding the relevant fields to the json file,
+but this would be a tedious process.
+
+There are some libraries,
+such as geopandas
+which can ameliorate this situation,
+but we could also use Altair directly to save time.
+
+To link two datasets together,
+Altair uses the `transform_lookup` method
+to *look up* the data column we want to use from the dataframe
+via a shared key column
+and add it to the plotted data.
+This process is similar to using `merge` in pandas.
+
+You could see in the previous slide that there exists an ID column
+and this contain the state IDs in alphabetical order,
+just like the geojson so we can use that for the lookup.
+
+The columns we want to fetch from the dataframe are passed in a list,
+and here we have specified only the population,
+which we use to color the states accordingly to how many people live in each state.
+
+---
+
+## Choropleths can be misleading since the area of each region matters
+
+```{python}
+us_map = (alt.Chart(state_map).mark_geoshape(color='lightgray', stroke='white')
+          .project(type='albersUsa'))
+points = alt.Chart(state_pop).mark_circle().encode(
+    longitude='longitude',
+    latitude='latitude',
+    size='population')
+(us_map + points).configure_view(stroke=None)
+```
+
+Notes:
+One drawback with choropleth maps is that
+they rely on the area of each region,
+even if it is unrelated to the measure we are studying.
+
+For example,
+just because the states in the midwest of the US have a large area,
+they are very prominent in the choropleth map we made on the last slide
+although their population is actually really small.
+
+On the other hand,
+if there was a state that had a small area,
+but a large population,
+we would not see it in the choropleth map.
+These issues are often discussed around the US election,
+where different approach are used to illustrate that the weight
+of each states vote is not tied to the area of the state.
+
+On effective way to convey this message is to use
+a map as a background
+and plot points on top that are scaled according to the measure of interest.
+
+To achieve this in Altair,
+we need one pair of coordinates to plot for each state,
+which we have in the column `latitude` and `longitude`
+(these represent the geometric center of each state).
+
+We then plot these coordinates on top of a map
+using the special `longitude` and `latitude` parameters
+and set the size of the points to be relative the population parameter.
+
+Finally,
+we remove the square grey outline by setting the stroke to `None`.
+
+Now it is much easier to see that many of the states in Eastern USA
+have a population several times that in the Midwest,
+although their area is smaller.
```

Comment:
Not clear why we are using the word special in the bottom part of this transcript, can you please clarify?


```suggestion
One drawback with choropleth maps is that
they rely on the area of each region,
even if it is unrelated to the measure we are studying.

For example,
just because the states in the midwest of the US have a large area,
they are very prominent in the choropleth map we made on the last slide
although their population is actually quite small.

On the other hand,
if there was a state that had a small area,
but a large population,
we might not even see it on the choropleth map.
These issues are often discussed around the US election,
where different approach are used to illustrate that the weight
of each states' vote is not tied to the geographical area of the state.

One effective way to convey this message is to use
a map as a background
and plot points on top that are scaled according to the measure of interest.

To achieve this in Altair,
we need one pair of coordinates to plot for each state. For this
we can use the columns `latitude` and `longitude` which represent the 
geometric centre of each state).

We then plot these coordinates on top of a map
using the special `longitude` and `latitude` parameters
and set the size of the points to be relative the population.

Finally,
we remove the square grey outline by setting the stroke to `None`.

Now it is much easier to see that many of the states in Eastern USA
have a population several times that of the states in the Midwest,
although their area is smaller.
```

Reply:
I guess "special" versus the normal x and y, but I agree it is moslty confusing, will remove.